hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -6043,7 +6079,7 @@ void updatePipeline(
       updatePipelineInternal(clientName, oldBlock, newBlock, newNodes,
           newStorageIDs, logRetryCache);
     } finally {
-      writeUnlock("updatePipeline");
+      writeUnlock(FSNamesystemLockMode.GLOBAL, "updatePipeline");
```

Comment:
@ZanderXu Hi, sir. Have a question here. In documentaion, it is said that updateBlockForPipeline and updatePipeline are only involved by blocks operations. Why we use FSNamesystemLockMode.GLOBAL here?

Reply:
That documentation is the final solution. Currently this PR is the 1st stage. This stage only aims to split the global lock into bm and fs locks.
In updatePipelineInternal we can see it uses inode too, so we use global lock here. Will solve similar cases in 2nd stage.
@ZanderXu  any other comments?

Reply:
Yes, this RPC involves iNode and FSEditLog, so we use global lock here.